clippy_lints/src/escape.rs

```diff
@@ -60,12 +61,18 @@ impl<'tcx> LateLintPass<'tcx> for BoxedLocal {
     fn check_fn(
         &mut self,
         cx: &LateContext<'tcx>,
-        _: intravisit::FnKind<'tcx>,
+        fn_kind: intravisit::FnKind<'tcx>,
         _: &'tcx FnDecl<'_>,
         body: &'tcx Body<'_>,
         _: Span,
         hir_id: HirId,
     ) {
+        if let Some(header) = fn_kind.header() {
+            if header.abi == Abi::C {
```

Comment:
I'm not familiar with FFI, but wouldn't it be safer to exit the function if the ABI is not `Rust`?

Reply:
I wasn't sure if other ABIs would depend on `Box` as well, so I limited to what the issue was telling. But checking against `Rust` seems to make sense.